cpp/src/arrow/filesystem/gcsfs.h

```diff
@@ -37,6 +37,9 @@ struct ARROW_EXPORT GcsOptions {
   std::string endpoint_override;
   std::string scheme;
 
+  /// Limits the number of concurrent operations, such as Object deletes.
+  int maximum_concurrent_operations = 64;
```

Comment:
No need for this if we use the IOContext's executor (the filesystem may decide to provide its own executor, or use the default IO thread pool).

Reply:
Removed.